### ISSUE TITLE
Enable secure boot on Windows 10 IoT Core

### ISSUE THREAD

**Issue by mike-tang0411** [ms-iot/security#20]
I installed the "KEK" and "db"  certificates to our UEFI firmware, And enabled UEFI secure boot mode. But the IoT core can't boot from the secure UEFI firmware.  Could you give me some advice on how to enable secure  boot on IoT core?

**Comment by IoTGirl** [comment]
Hi Mike,
There are quite a few documents on enabling Security on IOT Core:

- https://developer.microsoft.com/en-us/windows/iot/docs/turnkeysecurity

- https://developer.microsoft.com/en-us/windows/iot/docs/buildingsecuredevices

Sincerely,
IoTGirl

**Comment by LordOfDorks** [comment]
You have to install capsules for PK, KEK and DB as described in the documentation above, or UEFI secure boot will not be functional.

**Comment by mike-tang0411** [comment]
Thanks for the reply.
First of all, let me describe the situation I have encountered. I had installed the PK, KEK and DB to UEFI firmware during the UEFI boot process, the KEK and DB certificates are obtained from IoT GitHub (ms-iot/security/TurnkeySecurity/db/), and the PK certificate was created by myself. I can prove that these certificates have been installed successfully through the UEFI shell command. Then the Win IoT OS loader(bootarm.efi) was validated by the UEFI DB database , and I can see the Windows logo show on the screen, but the logo disappeared quickly and the device will restart.
According to the contents of the document above, I need to download DeviceLockDown Script package and generate required packages, then install four cab packages to IoT Core. I'd like to confirm the following questions : 
1. I can install the PK, KEK and DB in the IoT core system, so I don't need to install these certificates during UEFI boot process ? 
2. I just need to enable secure boot, Can I manually configure it by the FWVar command ?

**Comment by LordOfDorks** [comment]
If you can see the Windows Logo then the UEFI secure boot feature has successfully permitted the boot of the device and the reboot is triggered by a different reason. May I ask what device you are attempting to boot?


  1.  Correct, you can boot the device unprotected and then apply the capsules with FWVar.exe as described in the documentation.
  2.  Correct. UEFI secure boot can operate stand alone.

I added Alan Chan alanch@microsoft.com<mailto:alanch@microsoft.com> to this thread since he has more hands on knowledge and may have other insights for you.

Thank you
S.

**Comment by mike-tang0411** [comment]
My development board is Banana Pi BPI-M64， Which based on AllWinner 64-bit SoC.  I'm a UEFI engineer for AllWinner, Our team is porting the Win 10 IoT system to our Soc platform.

Update the issue status as below:

I built capsules with pre-generated keys and ‘QCDB_settings.xml’ file, Then booted the device unprotected and applied the capsules(db, KEK, PK) by Powershell command 'FWVar.exe '.  I can prove the  capsules were applied success by the UEFI Shell Command after the device reset, but  I encountered the situation as before: The os loader was validated by the UEFI firmware and the windows logo displayed on the screen, But the device can't boot to IoT system and restart after a period of time. In addition, I can't connect to the device by WinDbg tool through the Win IoT boot process. To summarize, I think the os loader can't boot the OS kernel.

Look forward to your suggestion on this issue.
